presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduConnectorFactory.java

```diff
@@ -59,6 +70,25 @@ public Connector create(String catalogName, Map<String, String> config, Connecto
                 .setRequiredConfigurationProperties(config)
                 .initialize();
 
-        return injector.getInstance(KuduConnector.class);
```

Comment:
we can inject `ClassLoaderSafeNodePartitioningProvider` directly so we can directly get Connector instance usiing injector.getInstance

Reply:
![Screenshot from 2020-06-10 13-53-42](https://user-images.githubusercontent.com/33892550/84232060-12bf8300-ab22-11ea-8749-4207bd5205b2.png)
After I modified as your suggestion it shows errors like the picture, can you help me with that? @Praveen2112  I have checked the code in hive and iceberg, they just new a Connector in ConnectorFactory#create()

Reply:
1) Explicit bindings are required and java.lang.ClassLoader is not explicitly bound.
  while locating java.lang.ClassLoader
    for the 2nd parameter of io.prestosql.plugin.base.classloader.ClassLoaderSafeNodePartitioningProvider.<init>(ClassLoaderSafeNodePartitioningProvider.java:42)
  at io.prestosql.plugin.kudu.KuduModule.configure(KuduModule.java:67)

2) Explicit bindings are required and java.util.List<io.prestosql.spi.session.PropertyMetadata<?>> is not explicitly bound.
  while locating java.util.List<io.prestosql.spi.session.PropertyMetadata<?>>
    for the 9th parameter of io.prestosql.plugin.kudu.KuduConnector.<init>(KuduConnector.java:64)
  at io.prestosql.plugin.kudu.KuduModule.configure(KuduModule.java:58)

Reply:
For injecting the classloader we can use this snippet 
```
                binder -> {
                    binder.bind(ClassLoader.class).toInstance(KafkaConnectorFactory.class.getClassLoader());
                    binder.bind(TypeManager.class).toInstance(context.getTypeManager());
                    binder.bind(NodeManager.class).toInstance(context.getNodeManager());
                });
```
and adding this module in the bootstrap Ref : We can check KafkaConnectorFactory 

And instead of passing `List<PropertyMetdata>` we can inject `KuduSessionProperties` and use `KuduSessionProperties#getSessionProperties` inside the KuduConnector

Reply:
@Praveen2112 the first problem was fixed, thank you!
the second problem, I have removed the KuduConnector#getSessionProperties method, and keep the inject in KuduModule#configure(), after that presto can started successfully, but when I using the session property kudu.grouped_execution, it show error below:
`No session properties found for catalog: kudu`

Reply:
I think I have understood your suggestion about the 2nd problem, KuduConnector.sessionProperties should be type of KuduSessionProperties, not a List<PropertyMetdata>. Then the second problem is resolved!
Thank you!  :)